### ISSUE TITLE
`CSVLogger` does not save validation loss and metrics when `validation_freq` is > 1 is passed to `model.fit`

### ISSUE THREAD

**Issue by salvador-dali** [keras-team/keras#14049]
**System information**  
- Have I written custom code (as opposed to using example directory):  Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Any
- TensorFlow backend (yes / no):  Yes
- TensorFlow version:  2.2.0
- Keras version:  2.3.0-tf
- Python version:  3.7.3
- CUDA/cuDNN version:  
- GPU model and memory:  

**Describe the current behavior**  
`CSVLogger` does not save validation loss and metrics when `validation_freq` is > 1 is passed to `model.fit`
**Describe the expected behavior**  
Save this information

**Code to reproduce the issue** 
```
from tensorflow.keras.callbacks import CSVLogger
def build_model():
  # build your model
  return model

def get_data():
  # read your data
  return X, Y


model = build_model()
X, Y = get_data()
model.fit(
    X, Y, batch_size=32, epochs=100, validation_split=0.1, shuffle=True,
    callbacks=[CSVLogger('file.csv')],
    validation_freq=20
)
```

With `validation_freq` validation loss/metric is not saved.

**Comment by Hemanthkumar2112** [comment]
try like this
`csv_logger = keras.callbacks.CSVLogger('file.csv',separator=",",append=True)`

`model.fit(X, Y, batch_size=32, epochs=100, validation_split=0.1,callbacks [csv_logger],validation_freq=20)`

**Comment by karlau7** [comment]
Any update on this? I have the same problem. 

@Hemanthkumar2112  
My code is written as you suggested, but the problem persists. It seems that the metrics written to the csv file is decided on the first epoch, so when validation_freq > 1 only the train metrics are recorded and the validation metrics are overlooked.

**Comment by Hemanthkumar2112** [comment]
hi dude @karlau7  validation data is used to  validate training data with no of epochs. If you mentioned validation freq, for example 
 validation_freq=10 in which it run or validate the validation data for every 10 epochs. my suggestion is don't mention any validation freq to a fit parameter. if you want try with validation frequency with [2,10,20]. reference: [keras](https://keras.io/api/models/model/#fit)

**Comment by sushreebarsa** [comment]
@salvador-dali Is this still an issue?
Could you please let us know if you have referred to the above [comment](https://github.com/keras-team/keras/issues/14049#issuecomment-629257739) and let us know if it helped ?
Please refer to this [link](https://keras.io/guides/training_with_built_in_methods) and try with the latest TF versions as older versions are not supported .
Thanks!

**Comment by sampathweb** [comment]
Thanks for the patience.  The issue is fixed in commit #17725
